docs/transactions.pug

```diff
@@ -46,7 +46,36 @@ block content
     operation in a transaction, you need to pass the `session` as an option.
 
     ```javascript
-    [require:transactions.*basic example]
```

Comment:
I like that these are now using async/await, but I don't like that this gets rid of our automated tests. I think it would be better to just convert `transactions.test.js` to use async/await, and then move it behind `test/es-next` like what we did for the `findOneAndUpdate()` tutorial:

* https://github.com/Automattic/mongoose/blob/0c7eab161c2746df64786238194f32ccd0aede25/test/es-next/findoneandupdate.test.es6.js
* https://github.com/Automattic/mongoose/blob/0c7eab161c2746df64786238194f32ccd0aede25/test/es-next.test.js